proto/src/determined/api/v1/config_policies.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+
+package determined.api.v1;
+option go_package = "github.com/determined-ai/determined/proto/pkg/apiv1";
+
+import "google/protobuf/struct.proto";
+
+// PutWorkspaceConfigPoliciesRequest is a complete replacement of config
+// policies for the workspace and workload type.
+message PutWorkspaceConfigPoliciesRequest {
+  // Specifies which workspace the config policies apply to. Use global API for
+  // global config policies.
+  int32 workspace_id = 1;
+  // Specifies which task type the config policies apply to: EXPERIMENT or NTSC.
+  string workload_type = 2;
+  // The config policies to use. Contains both invariant configs and constraints
+  // in yaml or json format.
+  string config_policies = 3;
+}
+
+// Response to PutWorkspaceConfigPolicies request.
+message PutWorkspaceConfigPoliciesResponse {
+  // The config policies saved. Contains both invariant configs and constraints
+  // in yaml or json format.
+  google.protobuf.Struct config_policies = 1;
+}
+
+// PutGlobalConfigPoliciesRequest is a complete replacement of global config
+// policies for the workload type.
+message PutGlobalConfigPoliciesRequest {
+  // Specifies which task type the config policies apply to: EXPERIMENT or NTSC.
+  string workload_type = 1;
+  // The config policies to use. Contains both invariant configs and constraints
+  // in yaml or json format.
+  string config_policies = 2;
+}
+
+// Response to PutGlobalConfigPolicies request.
+message PutGlobalConfigPoliciesResponse {
+  // The config policies saved. Contains both invariant configs and constraints
+  // in yaml or json format.
+  google.protobuf.Struct config_policies = 1;
+}
+
+// GetWorkspaceConfigPoliciesRequest is used to list all task config policies
+// for the workspace and task type.
+message GetWorkspaceConfigPoliciesRequest {
+  // Specifies which workspace the config policies apply to. Use global API for
+  // global config policies.
+  int32 workspace_id = 1;
+  // Specifies which task type the config policies apply to: EXPERIMENT or NTSC.
+  string workload_type = 2;
+}
+
+// Response to GetWorkspaceConfigPolicies request.
+message GetWorkspaceConfigPoliciesResponse {
+  // The current config policies saved. Contains both invariant configs and
+  // constraints in yaml or json format.
+  google.protobuf.Struct config_policies = 1;
+}
+
+// GetGlobalConfigPoliciesRequest is used to list all global task config
+// policies for the task type.
+message GetGlobalConfigPoliciesRequest {
+  // Specifies which task type the config policies apply to: EXPERIMENT or NTSC.
```

Comment:
Gah, I know I left a bunch of comments replacing "task type" with "workload type". It is kinda bothering me that the enum struct is called "workload type" when we refer it as a "task type", since technically tasks and workloads are slightly different things (you can submit a workload, and tasks get created for that workload). I think of a task as more something that requires resource allocation/ a container to run, like a trial or NTSC task. An experiment (I think) isn't necessarily a task.
This is all very trivial though, so feel absolutely free to ignore the comments that replace "task type" with "workload type"!

Reply:
That makes sense. I was using "task" because the feature is "task config policies" ... but you're right, we should refer to it as "workload type".